src/otx/cli/cli.py

```diff
@@ -0,0 +1,316 @@
+"""CLI entrypoints."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+import yaml
+from jsonargparse import ActionConfigFile, ArgumentParser, Namespace, namespace_to_dict
+from rich.console import Console
+
+from otx import OTX_LOGO, __version__
+from otx.cli.utils import get_otx_root_path
+from otx.cli.utils.help_formatter import CustomHelpFormatter
+from otx.cli.utils.jsonargparse import get_short_docstring
+
+if TYPE_CHECKING:
+    from jsonargparse._actions import _ActionSubCommands
+
+_ENGINE_AVAILABLE = True
+try:
+    from otx.engine import Engine
+except ImportError:
+    _ENGINE_AVAILABLE = False
+
+
+class OTXCLI:
+    """OTX CLI entrypoint."""
+
+    def __init__(self) -> None:
+        """Initialize OTX CLI."""
+        self.console = Console()
+        self.parser = self.init_parser()
+        self._subcommand_method_arguments: dict[str, list[str]] = {}
+        self.add_subcommands()
+        self.config = self.parser.parse_args(_skip_check=True)
+
+        self.subcommand = self.config["subcommand"]
+        self.run()
+
+    def init_parser(self) -> ArgumentParser:
+        """Initialize the argument parser for the OTX CLI.
+
+        Returns:
+            ArgumentParser: The initialized argument parser.
+        """
+        parser = ArgumentParser(
+            description="OpenVINO Training-Extension command line tool",
+            env_prefix="otx",
+            formatter_class=CustomHelpFormatter,
+        )
+        parser.add_argument(
+            "-v",
+            "--version",
+            action="version",
+            version=f"%(prog)s {__version__}",
+            help="Display OTX version number.",
+        )
+        return parser
+
+    def subcommand_parser(self, **kwargs) -> ArgumentParser:
+        """Returns an ArgumentParser object for parsing command line arguments specific to a subcommand.
+
+        Returns:
+            ArgumentParser: An ArgumentParser object configured with the specified arguments.
+        """
+        parser = ArgumentParser(
+            formatter_class=CustomHelpFormatter,
+            **kwargs,
+        )
+        parser.add_argument(
+            "-v",
+            "--verbose",
+            action="count",
+            help="Verbose mode. This shows a configuration argument that allows for more specific overrides. \
+                Multiple -v options increase the verbosity. The maximum is 2.",
+        )
+        parser.add_argument(
+            "-c",
+            "--config",
+            action=ActionConfigFile,
+            help="Path to a configuration file in json or yaml format.",
+        )
+        parser.add_argument(
+            "--data_root",
+            type=str,
+            help="Path to dataset root.",
+        )
+        parser.add_argument(
+            "--task",
```

Comment:
I thought they meant the same thing. Wouldn't it be task? I thought it would be 'anomaly_classification', 'anomaly_detection', 'anomaly_segmentation' based on anomalib (just like in OTX 1.5).

However, I'm in favor of splitting task, train type and would like to see it discussed in the future.

Reply:
We were discussing if `task` could be confusing to some users as it might indicate a full task implementation. It is instead defines the type of the task, so it might be an idea to use `task_type`. This was the idea. 

I agree with splitting `task_type`/`task` and `train_type`. Let's discuss this in a future PR